scripts/lib-carousel.js

```diff
@@ -0,0 +1,389 @@
+  + '</svg>';
+const HTML_ARROW_RIGHT = '<svg fill="rgb(217, 217, 217)" xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 600 600">\n'
+  + '<path d="M345.441,248.292L151.154,442.573c-12.359,12.365-32.397,12.365-44.75,0c-12.354-12.354-12.354-32.391,0-44.744L278.318,225.92L106.409,54.017c-12.354-12.359-12.354-32.394,0-44.748c12.354-12.359,32.391-12.359,44.75,0l194.287,194.284c6.177,6.18,9.262,14.271,9.262,22.366C354.708,234.018,351.617,242.115,345.441,248.292z"/>\n'
+  + '</svg>\n';
+
+let activeSlide = 1;
+let slideShow = false;
+let slideCount = 0;
+let touchStartX = 0;
+let touchEndX = 0;
+let touchRelX = 0;
+
+/**
+ * Increment active slide value
+ * @param direction
+ */
+const incrementActiveSlide = (direction = 1) => {
+  if (direction > 0) {
+    activeSlide = (activeSlide + 1) <= slideCount ? activeSlide + 1 : 1;
+  } else {
+    activeSlide = (activeSlide - 1) > 0 ? activeSlide - 1 : slideCount;
+  }
+  return activeSlide;
+};
+
+/**
+ * add attributes stored in key-value object to an element
+ * @param el
+ * @param attributes
+ */
+const addAttributes = (el, attributes) => {
+  Object.keys(attributes).forEach((key) => {
+    el.setAttribute(key, attributes[key]);
+  });
+};
+
+/**
+ * Activate/deactivate bottom bullets based on active slide
+ * @param bulletId
+ */
+const activateBullet = (bulletId) => {
+  const btnNav = document.querySelector('.dotted-nav');
+  [...btnNav.children].forEach((el) => {
+    const isActive = el.id === bulletId;
+    el.classList.toggle('active', isActive);
+    el.classList.toggle('inactive', !isActive);
+    el.setAttribute('aria-current', isActive ? 'true' : 'false');
+  });
+};
+
+/**
+ * Slides picture switcher
+ * @param slideId: format: slider-slide-${activeSlide}
+ */
+const activateSlide = (slideId) => {
+  const slider = document.querySelector('.slider');
+  Array.from(slider.children).forEach((el) => {
+    if (el.id === slideId) {
+      el.classList.replace('hide', 'show');
+      el.style.removeProperty('left');
+    } else {
+      el.classList.replace('show', 'hide');
+    }
+  });
+};
+
+/**
+ * Navigate with Bottom Bullet navigation
+ * @param event
+ */
+const dottedNavigation = (event) => {
+  const targetId = event.target.id.match(/\d+/)?.[0] || 1;
+  const sliderTarget = `slider-slide-${targetId}`;
+  activateSlide(sliderTarget);
+  activateBullet(event.target.id);
+  activeSlide = incrementActiveSlide();
+};
+
+/**
+ * Navigate with left and right for arrow navigation
+ * @param event
+ */
+const arrowNavigation = (event) => {
+  const navButton = event.currentTarget.id;
+  const increment = navButton === 'slider-arrow-left' ? -1 : 1;
+
+  activeSlide = incrementActiveSlide(increment);
+  activateSlide(`slider-slide-${activeSlide}`);
+  activateBullet(`slider-dot-${activeSlide}`);
+};
+
+/**
+ * Add event listeners for bottom bullet navigation
+ */
+const dottedNavOnClickEvents = () => {
```

Comment:
Use cache and do [event delegation](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#event_delegation):
```
const dottedNavOnClickEvents = () => {
  btnNav.addEventListener('click', (event) => {
    if (event.target && event.target.nodeName === 'BUTTON') {
      dottedNavigation(event);
    }
  });
};
```